source/common/config/subscription_factory_impl.cc

```diff
@@ -28,6 +28,13 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
   std::unique_ptr<Subscription> result;
   SubscriptionStats stats = Utility::generateStats(scope);
 
+  const auto transport_api_version = config.api_config_source().transport_api_version();
+  if (transport_api_version == envoy::config::core::v3::ApiVersion::V2 &&
+      runtime_.snapshot().runtimeFeatureEnabled("api.enable_deprecated_warning")) {
+    ENVOY_LOG(warn,
```

Comment:
I think this should also increment the deprecated stat like other deprecations? Can we do that?